### PR TITLE
Change Message When ROA Conflicts ROUTE

### PR DESCRIPTION
**Pull request by MiguelAHM**

### CODE REVIEW

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java [hunk 1]
```diff
@@ -2083,21 +2083,16 @@ public void search_route6_roa_mismatch_less_specific_as_xml_strings() {
...
                 "        </objectmessage>\n" +
                 "    </objectmessages>\n" +
                 "</object>\n" +
                 "</objects>\n" +
                 "<terms-and-conditions xlink:type=\"locator\" xlink:href=\"https://apps.db.ripe.net/docs/HTML-Terms-And-Conditions\"/>\n" +
-                "<version " +
-                "version=\"" + applicationVersion.getVersion() + "\" " +
-                "timestamp=\"" + applicationVersion.getTimestamp() + "\" " +
-                "commit-id=\"" + applicationVersion.getCommitId() + "\"/>\n" +
+                "<version version=\"0.1-TEST\" timestamp=\"20240610-1352\" commit-id=\"0\"/>\n" +
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java:2095]: Why are you hard-coding the variables? The `getTimestamp()` will change and fail the test. Use the `applicationVersion` instead (as before)

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/search/RpkiRoaMessageGenerator.java [hunk 2]
```diff
@@ -57,9 +57,9 @@ private RpslMessage validateRoa(final RpslObject rpslObject){
         }
 
         return switch (invalidRpkiRoa.getValue()) {
-            case INVALID_ORIGIN -> new RpslMessage(QueryMessages.roaRouteOriginConflicts(rpslObject.getType().getName(), invalidRpkiRoa.getKey().getMaxLength(), invalidRpkiRoa.getKey().getAsn()));
-            case INVALID_PREFIX_LENGTH -> new RpslMessage(QueryMessages.roaRoutePrefixLengthConflicts(rpslObject.getType().getName(), invalidRpkiRoa.getKey().getMaxLength(), invalidRpkiRoa.getKey().getAsn()));
-            case INVALID_PREFIX_AND_ORIGIN -> new RpslMessage(QueryMessages.roaRouteConflicts(rpslObject.getType().getName(), invalidRpkiRoa.getKey().getMaxLength(), invalidRpkiRoa.getKey().getAsn()));
+            case INVALID_ORIGIN -> new RpslMessage(QueryMessages.roaRouteOriginConflicts(rpslObject.getType().getName().toUpperCase(), invalidRpkiRoa.getKey().getMaxLength(), invalidRpkiRoa.getKey().getAsn()));
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/search/RpkiRoaMessageGenerator.java:60]: We are just specifying the prefix length in the error message, but shouldn't we be spelling out the **prefix** also with the length? It improves the clarity and usefulness I think

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpki/WhoisRoaChecker.java [hunk 3]
```diff
@@ -30,16 +31,12 @@ public Map.Entry<Roa, ValidationStatus> validateAndGetInvalidRoa(final RpslObjec
...
                 .stream()
                 .filter(getValidOrNotFoundRoas()).findFirst()
-                .or(() -> getInvalidRoas(roasStatus).findFirst());
-
-        if (roaStatusMap.isEmpty()){
-            return null;
-        }
-        return roaStatusMap.get();
+                .or(() -> getInvalidRoas(roasStatus).findFirst())
+                .orElse(null);
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/rpki/WhoisRoaChecker.java:39]: Add `Nullable` annotation to method if returning null value explicitly

Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/rpki/WhoisRoaChecker.java:39]: Are all callers checking for null? (YES I checked)

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/search/RpkiRoaMessageGenerator.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/rest/search/RpkiRoaMessageGenerator.java:60]: Riccardo was ok with the prefix.
so I should change this to put the prefix
We can also say something about the max length

Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/rest/search/RpkiRoaMessageGenerator.java:60]: Talk over zoom

**Review by eshryane — APPROVED** [review]

**Pull request merged**